azure/resource-containers/pf-resource-group/functions/public/new-pfresourcegroup.ps1

```diff
@@ -0,0 +1,14 @@
+Function New-PfResourceGroup
+{
+    if($global:_PfAzureContext)
+    {
+        $rg = [PfResourceGroupBuildState]::New($global:_PfAzureContext)
+    }
+    else{
+        $rg = [PfResourceGroupBuildState]::New()
+    }
+    $global:_PfDeploymentContext.ResourceContainers.Add($rg) | Out-Null
+    return $rg
```

Comment:
I have an array of "ResourceContainers" now off the deployment context.  I thought that this is the area we could try our first "hack" with dependencies?  We would only have Subscription, Resource Groups, and Management Groups to work with.  But this would allow us to get a design down before attacking resources, maybe?

Reply:
yeah, seems like should be able to implement routine that can generate a 'deployment plan' based on the dependencies between containers also

Reply:
Yes, I really like this idea.  Need to add this to our backlog